### ISSUE TITLE
Document is fetched multiple times while opening in "Correction"

### ISSUE THREAD

**Issue by aakashysharma** [webanno/webanno#182]
While opening a document in "Correction", I could see on console that it loads it fetches it couple of times. Why is this so?

I'm logged in with "admin" account.

```
2016-03-12 19:40:08 DEBUG [http-bio-8080-exec-11] [admin] (RepositoryServiceDbData) - Getting annotation document [Kobalt_SWE_001_2011_06_v1-5_ZH2.tsv] with ID [1] in project ID [1] for user [admin]
2016-03-12 19:40:08 DEBUG [http-bio-8080-exec-11] [admin] (RepositoryServiceDbData) - Getting annotation document [Kobalt_SWE_001_2011_06_v1-5_ZH2.tsv] with ID [1] in project ID [1] for user [CORRECTION_USER]
2016-03-12 19:40:09  INFO [http-bio-8080-exec-11] [admin] (RepositoryServiceDbData) - Upgraded CAS of user [CORRECTION_USER] for document [Kobalt_SWE_001_2011_06_v1-5_ZH2.tsv]  in project ID [1]
2016-03-12 19:40:09 DEBUG [http-bio-8080-exec-11] [admin] (RepositoryServiceDbData) - Getting annotation document [Kobalt_SWE_001_2011_06_v1-5_ZH2.tsv] with ID [1] in project ID [1] for user [admin]
2016-03-12 19:40:09 DEBUG [http-bio-8080-exec-11] [admin] (RepositoryServiceDbData) - Getting annotation document [Kobalt_SWE_001_2011_06_v1-5_ZH2.tsv] with ID [1] in project ID [1] for user [CORRECTION_USER]
2016-03-12 19:40:09 DEBUG [http-bio-8080-exec-11] [admin] (RepositoryServiceDbData) - Getting annotation document [Kobalt_SWE_001_2011_06_v1-5_ZH2.tsv] with ID [1] in project ID [1] for user [admin]
2016-03-12 19:40:09 DEBUG [http-bio-8080-exec-11] [admin] (RepositoryServiceDbData) - Getting annotation document [Kobalt_SWE_001_2011_06_v1-5_ZH2.tsv] with ID [1] in project ID [1] for user [CORRECTION_USER]
2016-03-12 19:40:09 DEBUG [http-bio-8080-exec-11] [admin] (RepositoryServiceDbData) - Getting annotation document [Kobalt_SWE_001_2011_06_v1-5_ZH2.tsv] with ID [1] in project ID [1] for user [admin]
2016-03-12 19:40:09 DEBUG [http-bio-8080-exec-11] [admin] (RepositoryServiceDbData) - Getting annotation document [Kobalt_SWE_001_2011_06_v1-5_ZH2.tsv] with ID [1] in project ID [1] for user [CORRECTION_USER]
2016-03-12 19:40:09 DEBUG [http-bio-8080-exec-10] [admin] (RepositoryServiceDbData) - Getting annotation document [Kobalt_SWE_001_2011_06_v1-5_ZH2.tsv] with ID [1] in project ID [1] for user [CORRECTION_USER]
2016-03-12 19:40:10  INFO [http-bio-8080-exec-2] [admin] (BratAnnotator) - AJAX-RPC CALLED: [whoami]
2016-03-12 19:40:10  INFO [http-bio-8080-exec-2] [admin] (BratAnnotator) - AJAX-RPC DONE: [whoami]
2016-03-12 19:40:10  INFO [http-bio-8080-exec-15] [admin] (BratAnnotator) - AJAX-RPC CALLED: [getCollectionInformation]
2016-03-12 19:40:10  INFO [http-bio-8080-exec-15] [admin] (BratAnnotator) - AJAX-RPC DONE: [getCollectionInformation]
2016-03-12 19:40:15 DEBUG [http-bio-8080-exec-12] [admin] (RepositoryServiceDbData) - Getting annotation document [Kobalt_SWE_001_2011_06_v1-5_ZH2.tsv] with ID [1] in project ID [1] for user [admin]
2016-03-12 19:40:15  INFO [http-bio-8080-exec-12] [admin] (BratAnnotator) - AJAX-RPC CALLED: [getDocument]
2016-03-12 19:40:15  INFO [http-bio-8080-exec-12] [admin] (BratAnnotator) - AJAX-RPC DONE: [getDocument]

```

**Comment by reckart** [comment]
We have had such things in the past on the annotation pages too - because the code was not streamlined, annotation data was read much more often than necessary. Optimally, the data should be read only once for the user and once for "CORRECTION_USER".

I only streamlined the annotation page itself at the time. Correction and automation modes are handled by @seyyaw .

**Comment by reckart** [comment]
Addressed via the cache implemented in #569.
